currencies/rate_converter.go

```diff
@@ -0,0 +1,165 @@
+package currencies
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"sync"
+	"time"
+)
+
+// RateConverter holds the currencies conversion rates dictionnary
+type RateConverter struct {
+	sync.RWMutex
+	httpClient            httpClient
+	updates               chan *Rates
+	done                  chan bool
+	ticker                *time.Ticker
+	fetchingInterval      time.Duration
+	retryFetchingInterval time.Duration
+	syncSourceURL         string
+	rates                 *Rates
+	lastUpdated           *time.Time
+}
+
+// NewRateConverter returns a new RateConverter
+func NewRateConverter(
+	httpClient httpClient,
+	syncSourceURL string,
+	fetchingInterval time.Duration,
+	retryFetchingInterval time.Duration,
+) *RateConverter {
+	return &RateConverter{
```

Comment:
We should fetch the initial set of currency data before returning this object. That way we know the object is ready to use before any callers try using it.

Reply:
yes, agree

Reply:
very good catch!

Reply:
Done